types/react-relay/lib/relay-experimental/EntryPointTypes.d.ts

```diff
@@ -33,18 +33,42 @@ export interface EnvironmentProviderOptions {
     [key: string]: unknown;
 }
 
-export interface PreloadedQuery<
+export type PreloadedQuery<
     TQuery extends OperationType,
     TEnvironmentProviderOptions = EnvironmentProviderOptions
-> {
-    readonly environment: IEnvironment;
-    readonly environmentProviderOptions: TEnvironmentProviderOptions | null | undefined;
-    readonly fetchKey: string | number | null | undefined;
-    readonly fetchPolicy: PreloadFetchPolicy;
-    readonly name: string;
-    readonly source: Observable<GraphQLResponse> | null | undefined;
-    readonly variables: TQuery['variables'];
-}
+> =
+    | PreloadedQueryInner_DEPRECATED<TQuery, TEnvironmentProviderOptions>
+    | PreloadedQueryInner<TQuery, TEnvironmentProviderOptions>;
+
+export type PreloadedQueryInner_DEPRECATED<
+    TQuery extends OperationType,
+    TEnvironmentProviderOptions = EnvironmentProviderOptions
+> = Readonly<{
+    environment: IEnvironment;
+    environmentProviderOptions: TEnvironmentProviderOptions | null | undefined;
+    fetchKey: string | number | null | undefined;
+    fetchPolicy: PreloadFetchPolicy;
+    name: string;
+    source: Observable<GraphQLResponse> | null | undefined;
+    variables: TQuery['variables'];
```

Comment:
This seems to be missing a few props: `kind`, `networkCacheConfig`, `id`, `status`.

Reply:
Off-topic, I’m surprised the linter didn’t complain about the use of `undefined` rather than marking the key as optional.

Reply:
Will amend, thanks!